src/nvim/option_defs.h

```diff
@@ -1080,4 +1081,31 @@ typedef struct vimoption {
 // buffers.  Indicate this by setting "var" to VAR_WIN.
 #define VAR_WIN ((char *)-1)
 
+// Option value type
+typedef enum {
+  kOptValTypeNil = 0,
+  kOptValTypeBoolean,
+  kOptValTypeNumber,
+  kOptValTypeString,
+} OptValType;
+
+// Option value
+typedef struct {
+  OptValType type;
+
+  union {
+    // Vim boolean options are actually tri-states because they have a third "None" value.
+    TriState boolean;
+    Integer number;
+    String string;
+  } data;
+} OptVal;
+
+EXTERN const char *optval_type_names[] INIT(= {
+  [kOptValTypeNil] = "Nil",
+  [kOptValTypeBoolean] = "Boolean",
+  [kOptValTypeNumber] = "Number",
+  [kOptValTypeString] = "String"
+});
```

Comment:
This is only used in option.c, so I think it's better to make this a static variable in option.c until it is uses somewhere else.

Reply:
There also doesn't seem to be a previous example of putting an `EXTERN` constant array in a header file.